include/lsst/afw/geom/Extent.h

```diff
@@ -353,6 +356,98 @@ typedef Extent<double,2> ExtentD;
 typedef Extent<double,2> Extent2D;
 typedef Extent<double,3> Extent3D;
 
+/**
+ *  Return the component-wise truncation (round towards zero).
+ *
+ *  In Python, this is available as both a free function and a method on ExtentD.
+ */
+template <int N>
+Extent<int,N> truncate(Extent<double,N> const & input);
+
+/**
+ *  Return the component-wise floor (round towards more negative).
+ *
+ *  In Python, this is available as both a free function and a method on ExtentD.
+ */
+template <int N>
+Extent<int,N> floor(Extent<double,N> const & input);
+
+/**
+ *  Return the component-wise floor (round towards more positive).
+ *
+ *  In Python, this is available as both a free function and a method on ExtentD.
+ */
+template <int N>
+Extent<int,N> ceil(Extent<double,N> const & input);
+
+#ifndef SWIG
+
+// Some operators below need to take ExtentBase arguments rather than Extent to
+// avoid ambiguous overloads (since some competing operators are defined as member
+// functions on ExtentBase).
+
+template <typename T, int N>
+Extent<T,N> operator*(T scalar, ExtentBase<T,N> const & rhs) {
+    return rhs * scalar;
+}
+
+template <int N>
+Extent<double,N> operator*(ExtentBase<int,N> const & lhs, double rhs) {
+    return Extent<double,N>(static_cast<Extent<int,N> const &>(lhs)) * rhs;
+}
+
+template <int N>
+void operator*=(ExtentBase<int,N> & lhs, double rhs) {
+    static_assert(N < 0, "In-place multiplication of Extent<int,N> by double will truncate.");
```

Comment:
Although maybe a comment to that effect would be useful? Also, I think "would truncate" would be better, since the truncation isn't actually happening.